### PR TITLE
feature(java-web): make java-web template look more like template and not like reference-app

### PR DESCRIPTION
**Pull request by Mugenor**
- remove unnecessary stuff from java-web template – downstream, wiremock, etc
- make it easy to update values for the java-web template

### CODE REVIEW

File: java/web/skeleton/Makefile [hunk 1]
```diff
@@ -101,25 +121,79 @@ push-functional: ## Push the Functional Docker image to an image registry
...
+	docker image push $(REGISTRY)/$(EXTENDED_TEST_PATH)/$(image_name)-extended:$(image_tag)
+
 .PHONY: deploy-functional
-deploy-functional: ## Deploy helm chart of the reference app to functional namespace
-	helm upgrade --install $(app_name)-dependencies ./helm-charts/ref-app-dependencies -n $(app_name)-functional --set registry=$(REGISTRY)/$(FAST_FEEDBACK_PATH) --atomic
-	helm upgrade --install $(app_name) ./helm-charts/ref-app -n $(app_name)-functional --set registry=$(REGISTRY)/$(FAST_FEEDBACK_PATH) --set domain=$(BASE_DOMAIN) --set appUrlSuffix="-functional" --set tag=$(image_tag) --atomic
-	helm list -n $(app_name)-functional ## list installed charts in the given tenant namespace
+deploy-functional: ## Deploy helm chart of the app to functional namespace
+	helm upgrade --install $(app_name) helm-charts/app -n $(tenant_name)-functional \
+		--set registry=$(REGISTRY)/$(FAST_FEEDBACK_PATH) \
```
Comment (tmcalves) [inline java/web/skeleton/Makefile:131]: maybe for another task, but having so many sets feels cluttered. I'd move to have a values file per environment. If you agree, we can create a task for it

File: java/web/skeleton/functional/src/test/java/io/cecg/reference/Stepdefs.java [hunk 2]
```diff
@@ -28,81 +28,26 @@ public class Stepdefs {
     private final String baseUri = SystemUtils.getEnvironmentVariable("SERVICE_ENDPOINT", "http://localhost:8080");
     private RequestSpecification request;
     private Response response;
-    private String unique_id;
```
Comment (tmcalves) [inline java/web/skeleton/functional/src/test/java/io/cecg/reference/Stepdefs.java:31]: thinking out loud, should we have single nft and functional for both java and go applications?

File: java/web/skeleton/Makefile [hunk 3]
```diff
@@ -27,52 +39,56 @@ p2p-image:
...
 p2p-functional: build-functional push-functional create-ns-functional deploy-functional ## Runs only functional helm tests
-	helm test $(app_name) -n $(app_name)-functional --filter name=reference-app-functional-test
-	kubectl logs reference-app-functional-test -n $(app_name)-functional
+	helm test $(app_name) -n $(namespace) --filter name=$(app_name)-functional-test
+	@echo Logs: $(call logs_url,"$(app_name)-functional-test",$(current_timestamp))
 
 .PHONY: p2p-nft
+p2p-nft: namespace = $(tenant_name)-nft
 p2p-nft: build-nft push-nft create-ns-nft deploy-nft ## Runs only NFT helm tests
+	helm test $(app_name) -n $(namespace) --filter name=$(app_name)-nft-test
```
[the remark this answers is not in the document]
Comment (Mugenor) [reply java/web/skeleton/Makefile:50]: I would propose to change it later.
I'll leave a comment in the task

File: java/web/skeleton/Makefile (hunk 1 above)
Comment (Mugenor) [reply java/web/skeleton/Makefile:131]: I'll include this point in the existing task

File: java/web/skeleton/functional/src/test/java/io/cecg/reference/Stepdefs.java (hunk 2 above)
Comment (Mugenor) [reply java/web/skeleton/functional/src/test/java/io/cecg/reference/Stepdefs.java:31]: Why?

File: java/web/skeleton/Makefile [hunk 4]
```diff
@@ -1,14 +1,26 @@
 projectDir := $(realpath $(dir $(firstword $(MAKEFILE_LIST))))
 os := $(shell uname)
 app_name = java-web
-image_tag = $(VERSION)
+tenant_name = reference-applications
```
[the remark this answers is not in the document]
Comment (Mugenor) [reply java/web/skeleton/Makefile]: fixed

File: java/web/skeleton/functional/src/test/java/io/cecg/reference/Stepdefs.java (hunk 2 above)
Comment (tmcalves) [reply java/web/skeleton/functional/src/test/java/io/cecg/reference/Stepdefs.java:31]: if they are the same functionality, would avoid duplication. would also make sure that both applications stay consistent

Comment (Mugenor) [reply java/web/skeleton/functional/src/test/java/io/cecg/reference/Stepdefs.java:31]: This code supposed to be used by the clients.
So if they want to write their app in java, most probably they would want to have tests also in java. The same for go

**Review by tmcalves — APPROVED** [review]

**Pull request merged**